libs/core/coroutines/include/hpx/coroutines/detail/context_generic_context.hpp

```diff
@@ -90,7 +90,7 @@ namespace hpx { namespace threads { namespace coroutines {
 
             void* allocate(std::size_t size) const
             {
-#if defined(_POSIX_VERSION)
+#if defined(_POSIX_VERSION) && !(defined(__ARM64_ARCH_8__) && defined(__APPLE__))
```

Comment:
Would you mind adding a comment in from if this line explaining that this condition excludes MacOS/M1 from using the posix `mmap`?

Reply:
Made requested adjustments addressing the clang-format, and added comment.  Is that sufficient?